### DISCUSSION TITLE
Trigger the Restart Extensions through commands

### DISCUSSION THREAD

**Discussion by sameemqureshi** [microsoft/vscode-discussions#2410] [Extension Development QnA] [answered]
![restatrt extensions](https://github.com/user-attachments/assets/c47cfa93-b5b2-49e5-83b5-67a4019105c0)
Hey all,
How to trigger the Restart Extension button programmatically through vscode.commands?

**Comment by isidorn** [comment, marked as answer]
There should be a command for this. Look at the vscode source code and you can find the id.
Then you can `executeCommand` that id to restart the extensions.

**Comment by sameemqureshi** [comment]
Hey ,
Is this the Same as  await vscode.commands.executeCommand('workbench.action.restartExtensionHost');
Actually , I am triggering this when my extension is updated , 
But I think that sometimes this isn't starting the Extension Host as the Updates are not applied.

**Reply by maximal** [reply]
Seems like this is not the same. After pressing “Restart Extensions” button, it (the button itself) disappears, yet when executing the command, the button stays.
